vllm_spyre/v1/core/scheduler.py

```diff
@@ -182,14 +183,20 @@ def schedule(self) -> "SchedulerOutput":
         To avoid additional specialization, some requests are held back from the
         base scheduler but are restored after.
         """
+        # unlock the current decode batch if no requests are in running queue
+        if len(self.running) == 0:
+            self.batch_is_locked = False
+            logger.debug("Unlocking the current decode batch as no requests "
+                         "are in running queue")
         # First purge the full waiting queue into our holdback queue, preserving
         # priority
         while self.waiting:
             self.holdback_queue.append(self.waiting.popleft())
 
         # Check if new requests can be scheduled.
         while self.holdback_queue:
-            if self.can_schedule(self.holdback_queue[0]):
+            if not self.batch_is_locked and self.can_schedule(
+                    self.holdback_queue[0]):
                 # Add request to the waiting queue
                 self.waiting.append(self.holdback_queue.popleft())
             else:
```

Comment:
shouldn't this be tested directly in the `can_schedule()` function? Maybe it can be the first tested condition, and return False directly if wrong